### DISCUSSION TITLE
Raspberry Pi 5 (or even 4 !!) as an extender ?

### DISCUSSION THREAD

**Discussion by sconcina** [OpenSageTV/Community#97] [SageTV Extenders (HD300,HD200)]
Have been using Sage for 18 years and not ready to give it up. I have two retired HD 100, 1 HD 200  and 1 HD 300 plus assorted laptop clients. Everything still works, but sometimes I wish that I had a safety net in case the extenders die.
I just received my Raspberry Pi 5 with 8GB and am blown away about the speed and responsiveness. It would make an awesome extender !! even though I got the fan case, for most office style use (programming) the box is dead quiet.
I just installed JDK 11, and everything went well. What next ? what path should I follow ? I am a complete newb in Java, but an experienced Python, Javascript and C++ programmer.
Looked in the old forums, but the lack of search is an issue.....
Any pointers would be welcome !!!!
Thanks,
Stefano

**Comment by TGsell1** [comment]
Stefano.  Not a direct answer to your question but Firesticks and the Shield devices do a pretty good jobas extenders using the ‘mini client’.   However if you could program a Pi to be an HD300 that would be awesome!Sent from my iPhoneOn Nov 26, 2023, at 8:24 PM, sconcina ***@***.***> wrote:﻿
Have been using Sage for 18 years and not ready to give it up. I have two retired HD 100, 1 HD 200  and 1 HD 300 plus assorted laptop clients. Everything still works, but sometimes I wish that I had a safety net in case the extenders die.
I just received my Raspberry Pi 5 with 8GB and am blown away about the speed and responsiveness. It would make an awesome extender !! even though I got the fan case, for most office style use (programming) the box is dead quiet.
I just installed JDK 11, and everything went well. What next ? what path should I follow ? I am a complete newb in Java, but an experienced Python, Javascript and C++ programmer.
Looked in the old forums, but the lack of search is an issue.....
Any pointers would be welcome !!!!
Thanks,
Stefano

**Comment by nyplayer** [comment]
I would look at a MINI PC instead of a PI ... there are great deals cyber monday ... Example below.

https://www.amazon.com/Beelink-Computer-Desktop-Display-Ethernet/dp/B0BVFXCJ2V/ref=sr_1_1?crid=2TZUC6NND5657&keywords=beelink%2Bn95&qid=1701103892&sprefix=beelink%2Bn95%2Caps%2C181&sr=8-1&ufe=app_do%3Aamzn1.fos.18ed3cb5-28d5-4975-8bc7-93deae8f9840&th=1

**Reply by sconcina** [reply]
Thanks, but I went down that path a few times already and have always been disappointed. The issue is the fan! none of these lower cost units are passively cooled, and they are severely underpowered, so that the moment they start running some significant load, the little bugger comes on and ruins the whole experience. Passively cooled devices are twice the price.

**Reply by nyplayer** [reply]
I have one of these ...

https://www.amazon.com/dp/B09XB3WL91?ref=nb_sb_ss_w_as-reorder_k0_1_9&amp=&crid=LYOLJMHXM27J&sprefix=mele%2Bmini&th=1

**Reply by sconcina** [reply]
Well, I stand corrected ! the unit you referenced looks perfectly adequate and even looks good, thanks !

**Comment by TGsell1** [comment]
These are really good.  I run two using WMC and cable card tuner.   I ‘freeze’ windows from updating.   Are you thinking to use these as a client? Would they show closed captions? Sent from my iPhoneOn Nov 27, 2023, at 11:52 AM, nyplayer ***@***.***> wrote:﻿
I would look at a MINI PC instead of a PI in there are great deals cyber monday ... Example below.
https://www.amazon.com/Beelink-Computer-Desktop-Display-Ethernet/dp/B0BVFXCJ2V/ref=sr_1_1?crid=2TZUC6NND5657&keywords=beelink%2Bn95&qid=1701103892&sprefix=beelink%2Bn95%2Caps%2C181&sr=8-1&ufe=app_do%3Aamzn1.fos.18ed3cb5-28d5-4975-8bc7-93deae8f9840&th=1

**Reply by sconcina** [reply]
How do you successfully prevent Windows from updating ? I have tried everything and managed to slow things down, but once in a while I get a call from the wife that the Sage is not working, and it is always Microsoft's fault. I need to move the whole thing to Linux....

**Reply by eelpout** [reply]
What version of Windows are you on? I haven't had any auto update issues with it in over a decade, though my server is still on Windows 10 since it's on an older AMD CPU. But I still get a couple here and there. 

Are the issues related to recording devices?

**Reply by sconcina** [reply]
CPU is about 10 years old, so it is Windows 10. Come to think of it, since it is now obsolete I guess the updates are going away :)

**Reply by nyplayer** [reply]
On Windows 10 Pro I have AUTOLOGIN then I have enabled to not restart for updates while users is logged in.... That way when I have time and not in use I just restart.

![image](https://github.com/OpenSageTV/Community/assets/16063354/222afc94-3e9a-48de-aca4-8420c66ff62b)

**Reply by sconcina** [reply]
Excellent idea, I will make sure that my setup is the same, thanks !

**Comment by nyplayer** [comment]
As far as Captions go I have yet to find a decoder including Microsofts that does CC in SageTV.

**Comment by TGsell1** [comment]
My mini pcs run win 8.1 pro so that I can run WMC on them.  None of that is still supported but doesn’t matter.  Works beautifully.  If I recall there’s a program to prevent updating just don’t recall exactly.  But I thought I should do it since I’m running a ‘non updated version of windows 8.1’ on those Minis.   Again. Works better than anything but to be fair it took time to set up.  Sent from my iPhoneOn Nov 27, 2023, at 1:46 PM, sconcina ***@***.***> wrote:﻿
How do you successfully prevent Windows from updating ? I have tried everything and managed to slow things down, but once in a while I get a call from the wife that the Sage is not working, and it is always Microsoft's fault. I need to move the whole thing to Linux....

**Comment by TGsell1** [comment]
Yep! Except the HD300! Etc. Sent from my iPhoneOn Nov 27, 2023, at 1:39 PM, nyplayer ***@***.***> wrote:﻿
As far as Captions go I have yet to find a decoder including Microsofts that does CC in SageTV.

**Comment by emveepee** [comment]
Since I have ported my yauiclient for NextPVR from Windows to Linux, including RPi OS in response to the first post I would recommend libmpv as your starting point.  I never tried with Java but with c+ it is very capable.  On my NUC the client seemed better on Linux than on Windows

Addressing a few other points.  

- NUC's as noted can run Linux do which avoids the Windows update concern.  Pure Linux even running on RPi's update all the time.
- SageTV clients don't need to depend on DirectShow decoders.  The issue is SageTV development is dead, @sconcina maybe consider a writing a general purpose SageTV client.
- I haven't got an RPi5 yet but my N100 NUC was far cheaper and that included 2.5 Ghz networking, an SSD and a Windows license.
- using (and developing) native clients for NextPVR on Windows,  Linux, RPi2-4 (with LibreELEC), Android, Roku and most recently Tizen the $20 ONN Android is as good a client as the others for playback, if wireless isn't an issue.
- Importantly Americans need to consider AC-4 decoding non DRM ATSC 3.0 for their clients.  There are ffmpeg hacks but they are inferrior to real support plus you need to factor in using the Schedules Direct app for DRM, native support means  
- My personal favourite client right now is the Android Rocktec G2 (relatively cheap if you get it on sale Newegg ) It has a PVR friendly BT remote with number keys, it runs cool, dual boots CoreELEC for native Kodi, has reasonable DV support.   It also has native AC-4

**Reply by sconcina** [reply]
Wow, lots of dense information there. It is going to take me a while to digest and explore, thanks !
Stefano

**Comment by jdaminc** [comment]
Also looking into a mini pc to replace a dead hd100 box... What is everyone using to control it? Mine would just be a dedicated SageTV box, with no real reason to connect a keyboard or mouse. remote only to simplify using it for wife and grandkids

**Comment by TGsell1** [comment]
I’m not currently using a mini as a server (using two w cable cards!) but in the past I’ve used a usb it receiver with a Harmony remote but I’m sure there are Bluetooth and better options now Sent from my iPhoneOn Apr 4, 2024, at 11:14 AM, jdaminc ***@***.***> wrote:﻿
Also looking into a mini pc to replace a dead hd100 box... What is everyone using to control it? Mine would just be a dedicated SageTV box, with no real reason to connect a keyboard or mouse. remote only to simplify using it for wife and grandkids

**Comment by jdaminc** [comment]
I actually still have 2 USBUIRT, and 2 Hauppauge remotes that I used many years ago that I found after asking the question.. guessing I could get by with those to start
